api/app/Http/Controllers/EntityController.php

```diff
@@ -222,14 +237,149 @@ public function deleteOne($id)
     public function deleteMany(Request $request)
     {
         $requestCollection = $request->data;
-        $ids = $this->getIds($requestCollection, $this->getKeyName(), $this->validateRequestRule);
-        $models = $this->getRepository()->findMany($ids);
+        $models = $this->findOrFailCollection($requestCollection);
+        foreach ($models as $model) {
+            $model->delete();
+        }
+        return $this->getResponse()->noContent();
+    }
 
-        if (count($ids) !== $models->count()) {
-            throw $this->notFoundManyException($ids, $models, $this->getKeyName());
+    /**
+     * @param $id
+     * @return BaseModel
+     */
+    protected function findOrNewEntity($id)
+    {
+        $this->validateId($id, $this->getModel()->getKeyName(), $this->getIdValidationRule());
+
+        try {
+            return $this->getModel()->findByIdentifier($id);
+        } catch (ModelNotFoundException $e) {
+            return $this->getModel()->newInstance();
         }
+    }
 
-        $this->getRepository()->deleteMany($models);
-        return $this->getResponse()->noContent();
+    /**
+     * @param $id
+     * @return BaseModel
+     */
+    protected function findOrFailEntity($id)
+    {
+        $this->validateId($id, $this->getModel()->getKeyName(), $this->getIdValidationRule());
```

Comment:
@Redjik we don't need it yet, however the goal of this is to make the method call simpler, as the `$this->validateId($id, $this->getModel()->getKeyName(), $this->getIdValidationRule());` is repeated in a lot of places, when it could just be `$this->validateId($id)`

I don't see a problem with moving it out of the trait, so there is no real reason not to make the change.